### PR TITLE
to support multiple cuda versions in published onnxruntime-training package 

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: to support multiple CUDA versions in published onnxruntime-training packages 

**Motivation and Context**
- name training python package onnxruntime-training 
- use local version for supported cuda versions
- publish nightly (development) packages in a private channel
- (TODO) one release package will be published in PYPI repository

### CODE REVIEW

File: onnxruntime/test/python/onnxruntime_test_ort_trainer.py [hunk 1]
```diff
@@ -257,14 +257,14 @@ def get_loaders(self):
...
 
-        # TODO: Remove this temporary fix when the issue https://github.com/pytorch/vision/issues/3549 is resolved
-        # Resource http://yann.lecun.com/exdb/mnist/ is not available
-        datasets.MNIST.resources = [
-            ('https://ossci-datasets.s3.amazonaws.com/mnist/train-images-idx3-ubyte.gz', 'f68b3c2dcbeaaa9fbdd348bbdeb94873'),
-            ('https://ossci-datasets.s3.amazonaws.com/mnist/train-labels-idx1-ubyte.gz', 'd53e105ee54ea40749a09fcbcd1e9432'),
-            ('https://ossci-datasets.s3.amazonaws.com/mnist/t10k-images-idx3-ubyte.gz', '9fb629c4189551a2d022fa330f9573f3'),
-            ('https://ossci-datasets.s3.amazonaws.com/mnist/t10k-labels-idx1-ubyte.gz', 'ec29112dd5afa0611ce80d1b7f02629c')
-        ]
+        # # TODO: Remove this temporary fix when the issue https://github.com/pytorch/vision/issues/3549 is resolved
```
Comment (suffiank) [inline onnxruntime/test/python/onnxruntime_test_ort_trainer.py:260]: Should this be deleted?

**Review by suffiank — APPROVED** [review]

File: onnxruntime/test/python/onnxruntime_test_ort_trainer.py (hunk 1 above)
Comment (liqunfu) [reply onnxruntime/test/python/onnxruntime_test_ort_trainer.py]: good point, undone my change

**Review by thiagocrepaldi — APPROVED** [review]
@snnn and I were discussing dropping support to python 3.6 because Numpy did it in January. It would be a good chance to drop in this PR too

**Comment by snnn** [comment]
Sorry the python 3.6 thing is still in discussion.  We just got some feedbacks from our customer and I guess we still need it for this release.

**Comment by snnn** [comment]
Hi @liqunfu and @thiagocrepaldi l found a few training pipelines are still using CUDA 10.1. Do you plan to upgrade them to CUDA 10.2 or 11.x? Pytorch also stopped supporting CUDA 10.1, we could do the same.

**Comment by liqunfu** [comment]
> Sorry the python 3.6 thing is still in discussion. We just got some feedbacks from our customer and I guess we still need it for this release.

@snnn I will keep python 3.6 for now.

**Comment by liqunfu** [comment]
> Hi @liqunfu and @thiagocrepaldi l found a few training pipelines are still using CUDA 10.1. Do you plan to upgrade them to CUDA 10.2 or 11.x? Pytorch also stopped supporting CUDA 10.1, we could do the same.

That is right. In this PR I added onnxruntime/tools/ci_build/github/linux/docker/Dockerfile.ubuntu_gpu_training which is for CUDA 11.1.1. CI for inference is still using 10.1 (in Dockerfile.ubuntu_gpu).

**Comment by snnn** [comment]
> Dockerfile.ubuntu_gpu

I think all the inference pipelines have moved to CUDA 10.2 or 11.0. So you don't need to make a copy of the Dockerfile.ubuntu_gpu file.

**Pull request merged**